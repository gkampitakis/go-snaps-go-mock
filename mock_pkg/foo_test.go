package mockpkg

import (
	"testing"

	"go-test/mock_pkg/internal"

	"github.com/golang/mock/gomock"
)

func TestFoo(t *testing.T) {
	ctrl := gomock.NewController(t)

	// Assert that Bar() is invoked.
	defer ctrl.Finish()

	m := NewMockFoo(ctrl)

	// Asserts that the first and only call to Bar() is passed 99.
	// Anything else will fail.
	m.
		EXPECT().
		Bar(internal.EqSnap(t)).
		Return(101)

	SUT(m)
}
