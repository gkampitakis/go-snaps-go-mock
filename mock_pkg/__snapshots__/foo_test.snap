
[TestFoo - 1]
int(99)
---
